models/base.py

```diff
@@ -196,8 +202,16 @@ def make_layer_dict(self, *args, **kwargs) -> LayerDictRaw:
     assert name_ctx.maker is self
     name_ctx.is_subnet_ctx = True
     res = self._subnet_func(*args, **kwargs)
-    Copy()(res, name="output")
-    return self._make_layer_dict_from_subnet_ctx(name_ctx)
+    # check if we receive more than one layer as return of a subnet
+    if isinstance(res, (Layer, type(List[Layer]))):
+      Copy()(res, name="output")
+      return self._make_layer_dict_from_subnet_ctx(name_ctx)
+    else:
+      # we return more than one layer (thus also working on other layers of the subnet, that are not output
+      # by convention: first layer is the output layer
+      assert isinstance(res, Tuple)
```

Comment:
Remove the assert.

Also, use `nest` here as I wrote before. (`nest.flatten` and then `nest.pack_sequence_as`. As I said, see other related code, e.g. in pytorch-to-returnn.)

Reply:
```suggestion
     res_flat = nest.flatten(res)
```